### ISSUE TITLE
Typescript srcmaps/stacktraces broken with meteor 1.5-beta.3 onwards

### ISSUE THREAD

**Issue by sdarnell** [meteor/meteor#8927]
I'm raising this issue in the main meteor/meteor repo as I don't think it is Typescript specific, and that it was a change in the bundler that broke things/exposed the problem.

The essence of the issue is that srcmaps and stacktraces are out, sometimes by a small amount but often by much more making it impossible to debug typescript apps or even rely on server side stack traces.

I have a github repo which is simply a `meteor create --example angular2-boilerplate` followed by addition of a trivial file, and a reference to that file that triggers exceptions on startup.
See https://github.com/sdarnell/srcmap-bug

The line numbers in the console output are wrong with meteor 1.5 (specifically meteor 1.5-beta.3).  The debug.ts file is simply:
```
1: console.trace('BaBoom');
2: export class Debug {
3:    public static Bang() {
4:        throw new Error("Boom!");
5:    }
6: }
7:
```
With Meteor 1.4.2.3, 1.4.3, 1.4.4, 1.4.4.1, 1.4.4.2, 1.4.4.3, 1.5-beta.2 the file and line numbers are:
 - BaBoom: at meteorInstall.both.debug.js (`both/debug.ts:1:9`)
 - Boom!: at Function.Debug.Bang (`both/debug.ts:4:15`)

In Meteor 1.5-beta3, 1.5 and 1.5.1 they are:
 - BaBoom: at meteorInstall.both.debug.js (`both/collections/demo.collection.ts:4:87`)
 - Boom!: at Function.Debug.Bang (`both/debug.ts:6:2`)

In the first case the file name is also wrong, and the second case the line number is out by 2 (and the column is wrong).

By means of binary chopping in on the problem, it appears that it is https://github.com/meteor/meteor/commit/e469eeec0537bd1f019a0d3cf242013830379c84 which caused the issue (just before 1.5-beta.3 was released).

I wasn't expecting it, but if I undo this part of the change that removes srcmaps things work again:
```diff
-    .replace(/\/\/# sourceMappingURL=[^\n]+/g, '//');
+    .replace(/\n\/\/# sourceMappingURL=[^\n]+/g, "");
```
The change clearly affects the number of physical lines, and presumably this is perturbing things, so unless the ultimate app.js.map is generated to understand that the additional lines are removed, it will be out by 1 line per file included in that specific bundle of js files.

@benjamn do you have any thoughts on why this only affects typescript plugin generated code, and not general JS or ecmascript converted code?

If we stop removing the line, will it knock out of step srcmaps elsewhere if compensating code has been added.

Originally derived from https://github.com/meteor/meteor/issues/8611 and split off into https://github.com/barbatus/typescript/issues/56.

**Comment by sdarnell** [comment]
After some more experimenting, the problem also affects regular Javascript files, in that their line numbers are off. But this isn't usually a problem because if you don't have any generated files that produce source maps, the `replace()` doesn't kick in.

So I believe Typescript is just one case that would knock files out of step, but presumably coffeescript or other pre-processors that output source maps would also trigger the problem.

**Comment by hwillson** [comment]
Thanks for your work on this @sdarnell - your fix has been merged, and will be coming in 1.5.2.

**Comment by sdarnell** [comment]
Thanks, I saw 1.5.2 beta 6 included it and tried it already. :)
